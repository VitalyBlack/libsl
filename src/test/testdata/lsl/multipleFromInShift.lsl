libsl "1.0.0";
library simple;

types {
   Int(int);
}

automaton A {
   var i: Int;

   state s1, s2, s3;

   shift (s1, s2) -> s3 (f);

   fun f(param: Int);
}