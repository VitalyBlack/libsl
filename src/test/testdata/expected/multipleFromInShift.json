{
  "name": "simple",
  "lslVersion": "1.0.0",
  "types": [
    {
      "name": "Int",
      "isPointer": false,
      "kind": "simple",
      "realType": {
        "name": "int",
        "isPointer": false,
        "kind": "real"
      }
    }
  ],
  "automata": [
    {
      "name": "A",
      "type": "Int",
      "constructorVariables": [],
      "variables": [
        {
          "name": "i",
          "type": "Int"
        }
      ],
      "states": [
        {
          "name": "s1",
          "kind": "SIMPLE"
        },
        {
          "name": "s2",
          "kind": "SIMPLE"
        },
        {
          "name": "s3",
          "kind": "SIMPLE"
        }
      ],
      "shifts": [
        {
          "from": "s1",
          "to": "s3",
          "functions": {
            "name": "f",
            "args": [
              "Int"
            ]
          }
        },
        {
          "from": "s2",
          "to": "s3",
          "functions": {
            "name": "f",
            "args": [
              "Int"
            ]
          }
        }
      ],
      "functions": [
        {
          "name": "f",
          "automaton": "A",
          "args": [
            {
              "name": "param",
              "type": "Int"
            }
          ],
          "contracts": [],
          "statements": []
        }
      ]
    }
  ]
}